ddl/placement_rule_test.go

```diff
@@ -194,6 +194,30 @@ func (s *testPlacementSuite) TestPlacementBuild(c *C) {
 				},
 			},
 		},
+		{
+			input: []*ast.PlacementSpec{
+				{
+					Role:        ast.PlacementRoleVoter,
+					Tp:          ast.PlacementAdd,
+					Replicas:    3,
+					Constraints: `["+  zone=sh", "-zone = bj"]`,
+				},
+				{
+					Role: ast.PlacementRoleVoter,
+					Tp:   ast.PlacementDrop,
+				},
+			},
```

Comment:
Also try dropping a different role with Add or Alter.

Reply:
And multiple DROPs.

Reply:
two more tests here, and some more tests in sql. A better error output also.